libs/blocks/z-pattern/z-pattern.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/*
+ * Z-Pattern - v5.1
+ */
+
+import { decorateBlockBg, decorateBlockDaa, getBlockSize } from '../../scripts/decorate.js';
+import media from '../media/media.js';
+
+export function decorateHeadline(header, size) {
+  const headingRow = header.parentElement;
+  headingRow.classList.add('heading-row');
+  headingRow.parentElement.classList.add('container');
+  const headerClass = (size === 'large') ? 'heading-XL' : 'heading-L';
+  header.classList.add(headerClass, 'headline');
+}
+
+function getOddRowsCount(rows) {
+  let zRowsOddCount = 0;
+  rows.forEach((row) => {
+    const firstCol = row.querySelector(':scope > div > div:first-of-type');
+    const rowIsOdd = firstCol.querySelector('h1, h2, h3, h4, h5, h6');
```

Comment:
The z-pattern should drafted with accessibility in mind. 
So the suggested doc standard is 
`[image | copy]`
`[image | copy]`
etc. 
If the author goes off the rails and flips the order, this will check how many 'rows' are 'odd' meaning, their copy is first and the author must be intentionally switching them. Thus we don't try to give it styles to z-pattern it up